File: IgxGrid CRUD Target/src/app/services/northwind-data.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class NorthwindDataService {
  constructor(private http: HttpClient) { }
  
  private readonly API_ENDPOINT = 'https://localhost:7244';
  
  /* Customers */

  // create
  public addCustomer(data: string) {
    return this.http.post(`${this.API_ENDPOINT}/Customer`, data)
  }

  // read
  public getCustomerData(): Observable<any> {
    return this.http.get(`${this.API_ENDPOINT}/Customer`);
  }

  // update
  public updateCustomer(data: any) {
    return this.http.put(`${this.API_ENDPOINT}/Customer/`, data);
  }

  // delete
  public deleteCustomer(id: string) {
    return this.http.delete(`${this.API_ENDPOINT}/Customer/${id}`);
  }


  /* Employees */

  // create
  public addEmployee(data: string) {
    return this.http.post(`${this.API_ENDPOINT}/Employee`, data)
  }

  // read
  public getEmployeeData(): Observable<any> {
    return this.http.get(`${this.API_ENDPOINT}/Employee`);
  }

  // update
  public updateEmployee(data: any) {
    return this.http.put(`${this.API_ENDPOINT}/Employee/`, data);
  }

  // delete
  public deleteEmployee(id: string) {
    return this.http.delete(`${this.API_ENDPOINT}/Employee/${id}`);
  }
}

File: IgbGrid CRUD Target WASM/Grid Test/Data/NorthwindDataService.cs
using System.Net.Http.Json;

namespace Grid_Test.Northwind
{
    public class NorthwindDataService
    {
        private readonly HttpClient http;

        private readonly string ApiEndpoint = "https://localhost:7244";

        public NorthwindDataService(HttpClient http)
        {
            this.http = http;
        }

        #region Employees
        // create
        public async Task<HttpResponseMessage> AddEmployee(EmployeesType data)
        {
            return await http.PostAsJsonAsync($"{this.ApiEndpoint}/Employee", data);
        }

        // read
        public async Task<EmployeesType[]?> GetEmployeeData()
        {
            return await http.GetFromJsonAsync<EmployeesType[]>($"{this.ApiEndpoint}/Employee");
        }

        // update
        public async Task<HttpResponseMessage> UpdateEmployee(EmployeesType data)
        {
            return await this.http.PutAsJsonAsync($"{this.ApiEndpoint}/Employee/", data);
        }

        // delete
        public async Task<HttpResponseMessage> DeleteEmployee(double? id)
        {
            return await this.http.DeleteAsync($"{this.ApiEndpoint}/Employee/{id}");
        }
        #endregion

        #region Customers
        // create
        public async Task<HttpResponseMessage> AddCustomer(CustomersType data)
        {
            return await http.PostAsJsonAsync($"{this.ApiEndpoint}/Customer", data);
        }

        // read
        public async Task<CustomersType[]?> GetCustomerData()
        {
            return await http.GetFromJsonAsync<CustomersType[]>($"{this.ApiEndpoint}/Customer");
        }

        // update
        public async Task<HttpResponseMessage> UpdateCustomer(CustomersType data)
        {
            return await this.http.PutAsJsonAsync($"{this.ApiEndpoint}/Customer/", data);
        }

        // delete
        public async Task<HttpResponseMessage> DeleteCustomer(string? id)
        {
            return await this.http.DeleteAsync($"{this.ApiEndpoint}/Customer/{id}");
        }
        #endregion
    }
}